### ISSUE TITLE
plugin_value_list_clone() in plugin.c logs that CollectInternalStats is broken 

### ISSUE THREAD

**Issue by charliesignalfx** [collectd/collectd#2108]
*   Version of collectd: 5.7
*   Operating system / distribution: CentOS 7 and Alpine Linux

## Expected behavior

When `CollectInternalStats` is `true`, collectd cycles through and does not log error messages indicating that a plugin is broken.

## Actual behavior
`plugin_value_list_clone` logs a message that there is a broken plugin.

```
[2016-12-14 21:53:16] Initialization complete, entering read-loop.
[2016-12-14 21:53:16] plugin_value_list_clone: Unable to determine interval from context for value list "d78ff1a9a5d5/collectd-write_queue/queue_length". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
[2016-12-14 21:53:16] plugin_value_list_clone: Unable to determine interval from context for value list "d78ff1a9a5d5/collectd-write_queue/derive-dropped". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
[2016-12-14 21:53:16] plugin_value_list_clone: Unable to determine interval from context for value list "d78ff1a9a5d5/collectd-cache/cache_size". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
[2016-12-14 21:53:26] plugin_value_list_clone: Unable to determine interval from context for value list "d78ff1a9a5d5/collectd-write_queue/queue_length". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
[2016-12-14 21:53:26] plugin_value_list_clone: Unable to determine interval from context for value list "d78ff1a9a5d5/collectd-write_queue/derive-dropped". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
[2016-12-14 21:53:26] plugin_value_list_clone: Unable to determine interval from context for value list "d78ff1a9a5d5/collectd-cache/cache_size". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
```
In the following steps to reproduce I use only the logfile plugin and CollectInternalStats.  Separately I have used other plugins `cpu`, `df`, `disk`, etc with `logfile` and `CollectInternalStats`.  `CollectInternalStats true` is the only configuration that will produce this result.  Setting it to `false` seems to alleviate the symptom.

## Steps to reproduce
*  Create the following config file (collectd.conf)
```
CollectInternalStats true

LoadPlugin logfile

<Plugin logfile>
  LogLevel "info"
  File "/var/log/signalfx-collectd.log"
  Timestamp true
  PrintSeverity false
</Plugin>
```

*   Pull latest collectd-5.7 source tar ball.
*   Unarchive
*   cd into unarchived directory
*   `$ ./configure`
*   `$ make install all`
*   `$ src/daemon/collectd -C ~/collectd.conf`
*   tail the configured log file
*  See the error messages
*  kill the collectd process
*  modify the config file setting `CollectInternalStats false`
*  Delete the log file
*  Restart collectd using the earlier statement
*  tail the configured log file
*  The statements are no longer reported indicating this is specific to CollectInternalStats
```
[2016-12-14 22:26:39] Initialization complete, entering read-loop.
```

**Comment by bluebob** [comment]
For what it's worth I'm also seeing issues with the CollectInternalStats. Just migrated (same configuration) from 5.6.2 to 5.7.0 and the systemctl status -l collectd is full of messages like:
Dec 20 11:41:08 abox.acme.com collectd[40751]: plugin_value_list_clone: Unable to determine interval from context for value list "abox.acme.com/collectd-write_queue/queue_length". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
Dec 20 11:41:08 abox.acme.com collectd[40751]: plugin_value_list_clone: Unable to determine interval from context for value list "abox.acme.com/collectd-write_queue/derive-dropped". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.
Dec 20 11:41:08 abox.acme.com collectd[40751]: plugin_value_list_clone: Unable to determine interval from context for value list "abox.acme.com/collectd-cache/cache_size". This indicates a broken plugin. Please report this problem to the collectd mailing list or at <http://collectd.org/bugs/>.

Strange thing though is that my metrics store is still seeing non-zero (/reasonable) values for CacheSize and QueueLength, so either this is an intermittent "fault" or it's a false alarm.

**Comment by charliesignalfx** [comment]
@octo, I am not sure how I messed up in validating this, but I must have pulled down the wrong branch originally.  

This proposed fix _does not_ correct the issue I reported.  

To be sure, in a fresh ubuntu container I pulled down collectd:collectd-5.7 added the fix that you've proposed, cleaned, built, configured,  `make install all`ed and ran with the config I referenced earlier.  I still see the same broken plugin statements.
